src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
-    cardLikeButton.classList.toggle("card__like-button_liked");
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+
+    if (isLiked) {
+      api
+        .deleteLike(cardID)
+        .then((updatedCard) => {
+          cardLikeButton.classList.remove("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error(`Error: ${err}`);
+        });
+    } else {
+      api
+        .addLike(cardID)
+        .then((updatedCard) => {
+          cardLikeButton.classList.add("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error(`Error: ${err}`);
+        });
+    }
   });
 
   cardDeleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    currentCardElement = cardElement;
+    currentCardID = cardID;
+    openModal(deleteModal);
   });
 
   cardImage.addEventListener("click", () => {
-    cardModalCaption.textContent = data.name;
-    cardModalImage.src = data.link;
-    cardModalImage.alt = data.name;
-    openModal(cardModal);
+    cardElements.modalCaption.textContent = data.name;
+    cardElements.modalImage.src = data.link;
+    cardElements.modalImage.alt = data.name;
+    openModal(cardElements.modal);
   });
 
   return cardElement;
 }
 
-cardModalCloseButton.addEventListener("click", () => {
-  closeModal(cardModal);
+// Edit Avatar Modal
+const avatarModal = document.querySelector("#edit-avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarProfileButton = document.querySelector(".profile__avatar-button");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarInput = avatarModal.querySelector("#edit-avatar-input-link");
+const profileAvatarImage = document.querySelector(".profile__avatar");
+
+avatarProfileButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarModalCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+avatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const submitButton = avatarForm.querySelector(".modal__submit-button");
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  const avatarUrl = avatarInput.value;
+
+  api
+    .editAvatar(avatarUrl)
+    .then((data) => {
+      profileAvatarImage.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
+});
+
+// Delete Modal
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalForm = document.querySelector("#delete-modal-form");
+const cancelButton = document.querySelector(".modal__cancel-button");
+const closeButton = document.querySelector("#close-delete-modal");
```

Comment:
`closeButton`

There are 3 modals with different buttons and forms in the project. You need to **name** variables and functions the way all could understand what they are.


`confirmationCloseButton`